File: sbar/actor/item/window.lua
local palette = require 'actor.helper.color'

local window_title = BAR.add('item', 'window_title', {
	width = 'dynamic',
	position = 'right',
	background = { border_color = palette.blue },
	label = { color = palette.blue },
})

BAR.add('event', 'window_title_changed')
BAR.add('event', 'window_focused')

local yabai = require 'actor.helper.yabai'
window_title:subscribe({ 'window_focused', 'window_title_changed' }, function(env)
	if yabai.display.focused().index == DISPLAY_INDEX then
		local focused_window_title = yabai.window.focused().title
		window_title:set { label = focused_window_title }
	end
end)

File: sbar-rs/src/items/window.rs
use crate::helpers::colors::Colors;
use crate::helpers::icons::Icons;
use crate::helpers::yabai::DisplayInfo;
use crate::sketchybar::SketchyBar;
use anyhow::Result;

pub async fn setup(bar: &mut SketchyBar, display_info: &DisplayInfo,) -> Result<(),> {
	println!("🪟 Setting up window item");

	// Add window item
	bar.add("item", "window", "left",).await?;

	// Configure window properties
	bar.set(
		"window",
		&[
			("width", "dynamic",),
			("position", "left",),
			("icon", Icons::WINDOW,),
			("icon.color", &format!("0x{:08x}", Colors::GREEN),),
			("label", "Window",),
			("label.color", &format!("0x{:08x}", Colors::GREEN),),
			("background.border_color", &format!("0x{:08x}", Colors::GREEN),),
			("associated_display", &display_info.index.to_string(),),
		],
	)
	.await?;

	// Subscribe to window events
	bar.subscribe("window", &["window_focus", "window_title",],).await?;

	println!("✅ Window item configured");
	Ok((),)
}
